src/main/java/com/ibm/etcd/client/StaticEtcdNameResolverFactory.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2017, 2018 IBM Corp. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy
+ * of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.ibm.etcd.client;
+
+import java.net.URI;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import io.grpc.Attributes;
+import io.grpc.EquivalentAddressGroup;
+import io.grpc.NameResolver;
+import io.grpc.Status;
+import io.grpc.internal.DnsNameResolverProvider;
+
+/**
+ * NameResolver for static list of URIs. Delegates individual address resolution
+ * to gRPC DnsNameResolver.
+ */
+class StaticEtcdNameResolverFactory extends NameResolver.Factory {
```

Comment:
is `NameResolverProvider` preferred to `NameResolver.Factory`?

Reply:
I don't think so - a `Factory` the grpc channel builder expects. `NameResolverProvider` just happens to subclass it.